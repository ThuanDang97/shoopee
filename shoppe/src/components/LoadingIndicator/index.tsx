import React, { memo } from 'react'
import { Flex, Spinner, SpinnerProps } from '@chakra-ui/react'

const LoadingIndicator = ({ ...props }: SpinnerProps) => {
  return (
    <Flex
      alignItems="flex-start"
      justifyContent="center"
      backgroundColor="lightSilver"
      opacity={0.5}
      position="fixed"
      backdropFilter="blur(3px)"
      inset={0}
      zIndex={1500}
      paddingTop="150px"
    >
      <Spinner {...props} />
    </Flex>
  )
}

export default memo(LoadingIndicator)
