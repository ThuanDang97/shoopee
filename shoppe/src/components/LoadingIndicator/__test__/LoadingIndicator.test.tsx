import { render } from '@utils/testUtils'

import LoadingIndicator from '@components/LoadingIndicator'

describe('LoadingIndicator component', () => {
  it('should be renders loading indicator snapshot', () => {
    const { container } = render(<LoadingIndicator />)

    expect(container).toMatchSnapshot()
  })
})
