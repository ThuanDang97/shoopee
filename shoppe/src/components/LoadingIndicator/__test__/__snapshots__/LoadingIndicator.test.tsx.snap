// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingIndicator component should be renders loading indicator snapshot 1`] = `
<div>
  <div
    class="css-dvqovb"
  >
    <div
      class="chakra-spinner css-nkpswe"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
